examples/Migration/CampaignReportToCsv.cs

```diff
@@ -0,0 +1,117 @@
+
+        /// <summary>
+        /// Returns a description about the code example.
+        /// </summary>
+        public override string Description =>
+            "This code example illustrates how to get metrics about a campaign and serialize the " +
+            "result as a CSV file.";
+
+        /// <summary>
+        /// Runs the code example.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        public void Run(GoogleAdsClient client, long customerId,
+            string outputFilePath = OUTPUT_FILE_PATH)
+        {
+            GoogleAdsServiceClient googleAdsServiceClient =
+                client.GetService(Services.V8.GoogleAdsService);
+
+            // Create a query that retrieves campaigns.
+            string query = @"
+                SELECT
+                    campaign.id,
+                    campaign.name,
+                    segments.date,
+                    metrics.impressions,
+                    metrics.clicks,
+                    metrics.cost_micros 
+                FROM campaign 
+                WHERE segments.date DURING LAST_30_DAYS 
+                    AND campaign.status = 'ENABLED' 
+                ORDER BY segments.date DESC";
+
+            // Issues a search request with a specified page size.
+            SearchGoogleAdsRequest searchGoogleAdsRequest = new SearchGoogleAdsRequest
+            {
+                CustomerId = customerId.ToString(),
+                Query = query,
+                PageSize = PAGE_SIZE
+            };
+            PagedEnumerable<SearchGoogleAdsResponse, GoogleAdsRow> pagedResponse =
+                googleAdsServiceClient.Search(searchGoogleAdsRequest);
+
+            if (pagedResponse.Count() == 0)
+            {
+                Console.WriteLine("No results found!");
+                return;
+            }
+
+            StringBuilder csvRows = new StringBuilder();
+
+            // Set the header for the CSV file.
+            csvRows.AppendLine("Campaign.Id,Campaign.Name,Segments.Date,Metrics.Impressions," +
+                "Metrics.Clicks,Metrics.CostMicros");
+
+            // Iterate over all returned rows in all pages and extract the information.
+            foreach (GoogleAdsRow googleAdsRow in pagedResponse)
+            {
+                csvRows.AppendLine(string.Format("{0},{1},{2},{3},{4},{5}",
+                    googleAdsRow.Campaign.Id,
+                    googleAdsRow.Campaign.Name,
+                    googleAdsRow.Segments.Date,
+                    googleAdsRow.Metrics.Impressions,
+                    googleAdsRow.Metrics.Clicks,
+                    googleAdsRow.Metrics.CostMicros));
+            }
+
+            if (outputFilePath == null)
+            {
+                outputFilePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) +
```

Comment:
We can write this to home directory too. Putting a timestamp on the filename can be helpful. If the golden CL doesn't handle this, then we should make the change and ask others to fix accordingly.

Reply:
Updated.